### ISSUE TITLE
Are there any plans to release VAE training code?

### ISSUE THREAD

**Issue by ebartrum** [Lightricks/LTX-Video-Trainer#3]
Many thanks for releasing this impressive work!

I am currently using LTX Video as a base model for a research project.
For my specific use case, I need to modify and retrain the VAE, and have been having
some trouble getting this to work. In particular, my retrained VAE often outputs videos which are stationary,
with all video frames identical to the input image.
I wondered if you have observed this kind of failure case before, or have any insights on it?
Are you planning to release VAE training code at some point?

Thanks again and congratulations on the paper!

**Comment by guyltrks** [comment]
Hi Ed.

We did not observe this kind of failure case. Please notice the videos fed into the VAE should have 1+8*k frames for it to work properly. Otherwise, I would suspect an issue in the training code or data.

Just to make sure - this issue happens when reconstructing videos with your retrained (finetuned?) VAE, right?

**Comment by ebartrum** [comment]
Would you be able to release code for the Video DWT Loss mentioned in your paper section 2.1.5? That would be very helpful!

**Comment by ebartrum** [comment]
Correct, I haven't seen it with the original VAE, although I see other artifacts in that case! After going through the decoder code, I got the impression that the VideoUpsampler3d blocks are indexing the hidden states in such a way that token 0 only represents frame 0 (as mentioned in the caption to figure 4) is that right? whereas the other tokens represent 8 frames? I suspect that might be the underlying reason for the effect I have observed - but still need to test this

**Comment by harelc** [comment]
Yes, that is correct.  The starting frame is the only one with a 1:1
mapping to latents,  thereafter it's 8:1.

Harel Cain
Researcher
✧

Bridging the gap between imagination and creation
[image: photo]

[image: facebook] <https://www.facebook.com/Lightricks.Global/>

[image: instagram] <https://www.instagram.com/lightricks/>

[image: linkedin] <https://www.linkedin.com/company/lightricks>

[image: __tpx__]

בתאריך יום ה׳, 27 במרץ 2025, 14:52, מאת Ed Bartrum ‏<
***@***.***>:

**Comment by ebartrum** [comment]
I have additionally noticed an issue where the autoencoded frame 0 is noisier than the subsequent frames (again, this is in my trained LTX video autoencoder results). Do you have any insights into what may be causing that?

**Comment by xylshawn** [comment]
Your paper mentioned that Video DWT Loss and gan loss can ensure the reconstruction of high-frequency details, which are important for VAE training. Thus, are you planning to release VAE training code？
